### PR TITLE
Add Android v7 requirement API 19

### PR DESCRIPTION
**Pull request by stefanosiano**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

SDK v7 requires Android API level 19+.
This info has been added to the migration guide. Also, previous workaround for NDK on API 14 has been removed.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/platforms/android/migration.mdx [hunk 1]
```diff
@@ -6,6 +6,23 @@ redirect_from:
...
+```xml {filename:AndroidManifest.xml}
+<manifest>
+    <!-- Merging strategy for the imported manifests -->
+    <uses-sdk tools:overrideLibrary="io.sentry.android" />
+</manifest>
+```
+
+    With these changes the SDK will be enabled on all devices with API level ≥ 14 and the SDK with NDK will be enabled on all devices with API level ≥ 16.
+
+- Older versions without NDK integration support API level 14.
```
Comment (shanamatthews) [inline src/platforms/android/migration.mdx:24]:
```suggestion
- `io.sentry:sentry-android 7.0.0` requires API level 19.

    Older versions with the NDK integration support API level 16. To support devices on an API level lower than 16, update your `AndroidManifest.xml`:

    ```xml {filename:AndroidManifest.xml}
    <manifest>
        <!-- Merging strategy for the imported manifests -->
        <uses-sdk tools:overrideLibrary="io.sentry.android" />
    </manifest>
    ```

    With these changes the SDK will be enabled on all devices with API level ≥ 14 and the SDK with NDK will be enabled on all devices with API level ≥ 16.

- Older versions without the NDK integration support API level 14.
```

Comment (shanamatthews) [reply src/platforms/android/migration.mdx:24]: formatting tweaks and some small language tweaks

**Review by shanamatthews — APPROVED** [review]

**Comment by shanamatthews** [comment]
Hey @stefanosiano - is this PR ready for merge?

**Comment by stefanosiano** [comment]
> Hey @stefanosiano - is this PR ready for merge?

Hey! This pr is ready to be merged by itself, as it was reviewed, but we are waiting for it.
I marked this pr with the v7 milestone: when the Android SDK v7 will be released, I'll merge this (and others waiting for the same release).
I'm converting to draft for the time being. 👍

**Comment by shanamatthews** [comment]
> > Hey @stefanosiano - is this PR ready for merge?
> 
> Hey! This pr is ready to be merged by itself, as it was reviewed, but we are waiting for it. I marked this pr with the v7 milestone: when the Android SDK v7 will be released, I'll merge this (and others waiting for the same release). I'm converting to draft for the time being. 👍

Perfect, thanks!

**Pull request merged**